libs/langchain/langchain/schema/runnable/retry.py

```diff
@@ -0,0 +1,250 @@
+        config: Optional[Union[RunnableConfig, List[RunnableConfig]]] = None,
+        *,
+        return_exceptions: bool = False,
+        **kwargs: Any
+    ) -> List[Output]:
+        return self._batch_with_config(
+            self._batch, inputs, config, return_exceptions=return_exceptions, **kwargs
+        )
+
+    async def _abatch(
+        self,
+        inputs: List[Input],
+        run_manager: List[AsyncCallbackManagerForChainRun],
+        config: List[RunnableConfig],
+    ) -> List[Union[Output, Exception]]:
+        results_map: Dict[int, Output] = {}
+
+        def pending(iterable: List[U]) -> List[U]:
+            return [item for idx, item in enumerate(iterable) if idx not in results_map]
+
+        try:
+            async for attempt in self._async_retrying():
+                with attempt:
+                    # Get the results of the inputs that have not succeeded yet.
+                    result = await super().abatch(
+                        pending(inputs),
+                        self._patch_config_list(
+                            pending(config), pending(run_manager), attempt.retry_state
+                        ),
+                        return_exceptions=True,
+                    )
+                    # Register the results of the inputs that have succeeded.
+                    first_exception = None
+                    for i, r in enumerate(result):
+                        if isinstance(r, Exception):
+                            if not first_exception:
+                                first_exception = r
+                            continue
+                        results_map[i] = r
+                    # If any exception occurred, raise it, to retry the failed ones
+                    if first_exception:
+                        raise first_exception
+                if (
+                    attempt.retry_state.outcome
+                    and not attempt.retry_state.outcome.failed
+                ):
+                    attempt.retry_state.set_result(result)
+        except RetryError as e:
+            try:
+                result
+            except UnboundLocalError:
+                result = cast(List[Output], [e] * len(inputs))
+
+        outputs: List[Union[Output, Exception]] = []
+        for idx, _ in enumerate(inputs):
+            if idx in results_map:
+                outputs.append(results_map[idx])
+            else:
+                outputs.append(result.pop(0))
+        return outputs
+
+    async def abatch(
+        self,
+        inputs: List[Input],
+        config: Optional[Union[RunnableConfig, List[RunnableConfig]]] = None,
+        *,
+        return_exceptions: bool = False,
+        **kwargs: Any
+    ) -> List[Output]:
+        return await self._abatch_with_config(
+            self._abatch, inputs, config, return_exceptions=return_exceptions, **kwargs
+        )
+
+    # stream() and transform() are not retried because retrying a stream
```

Comment:
should these instead raise exception when called?

Reply:
I think no.. but open to have it change down the road